### ISSUE TITLE
client error: cannot patch basefile and failed to sync project...due to unhandled exception

### ISSUE THREAD

**Issue by rjwillson** [lutraconsulting/qgis-mergin-plugin#279]
for the last 5 or so days I have been getting these messages regularly when I try to sync using the QGIS plugin. It has now happened with 3 different projects. Initially, I thought maybe it was happening because my local files were stored in OneDrive. So I moved them to a local non-onedrive folder but unfortuanately the error messages continue. The only way I can fix the problem is to use mergin to remove the local copy and then re-download the project from the mergin server. Because two of the projects are 500 mb in size this is taking a lot of time and is not viable in long term partly because I can't predict when this is going to happen. The syncing will be okay for a day or so and I can sync several times. Then for some reason the error messages arrives and that's it. I haven't really changed anything on my end recently that I can think of.

Details below:
![2021-07-20 mergin plugin error messages](https://user-images.githubusercontent.com/11098144/126379779-789aeb82-7195-4e1c-b1e5-454ac58c567b.png)

```
Traceback (most recent call last):

  File "C:/Users/robwi/AppData/Roaming/QGIS/QGIS3\profiles\2019-10-27 New Profile for Q 3.1x/python/plugins\Mergin\mergin\merginproject.py", line 437, in apply_pull_changes
    self.geodiff.create_changeset(basefile, src, server_diff)

  File "C:/Users/robwi/AppData/Roaming/QGIS/QGIS3\profiles\2019-10-27 New Profile for Q 3.1x/python/plugins\Mergin\mergin\deps\pygeodiff\main.py", line 65, in create_changeset
    return self.clib.create_changeset(base, modified, changeset)

  File "C:/Users/robwi/AppData/Roaming/QGIS/QGIS3\profiles\2019-10-27 New Profile for Q 3.1x/python/plugins\Mergin\mergin\deps\pygeodiff\geodifflib.py", line 200, in create_changeset
    _parse_return_code(res, "createChangeset")

  File "C:/Users/robwi/AppData/Roaming/QGIS/QGIS3\profiles\2019-10-27 New Profile for Q 3.1x/python/plugins\Mergin\mergin\deps\pygeodiff\geodifflib.py", line 41, in _parse_return_code
    raise GeoDiffLibError(msg)

Mergin.mergin.deps.pygeodiff.geodifflib.GeoDiffLibError: createChangeset


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "C:/Users/robwi/AppData/Roaming/QGIS/QGIS3\profiles\2019-10-27 New Profile for Q 3.1x/python/plugins\Mergin\sync_dialog.py", line 306, in pull_timer_tick
    self.pull_conflicts = pull_project_finalize(self.job)

  File "C:/Users/robwi/AppData/Roaming/QGIS/QGIS3\profiles\2019-10-27 New Profile for Q 3.1x/python/plugins\Mergin\mergin\client_pull.py", line 555, in pull_project_finalize
    conflicts = job.mp.apply_pull_changes(job.pull_changes, job.temp_dir)

  File "C:/Users/robwi/AppData/Roaming/QGIS/QGIS3\profiles\2019-10-27 New Profile for Q 3.1x/python/plugins\Mergin\mergin\merginproject.py", line 445, in apply_pull_changes
    self.geodiff.make_copy_sqlite(f_conflict_file, dest)

  File "C:/Users/robwi/AppData/Roaming/QGIS/QGIS3\profiles\2019-10-27 New Profile for Q 3.1x/python/plugins\Mergin\mergin\deps\pygeodiff\main.py", line 221, in make_copy_sqlite
    return self.clib.make_copy_sqlite(src, dst)

  File "C:/Users/robwi/AppData/Roaming/QGIS/QGIS3\profiles\2019-10-27 New Profile for Q 3.1x/python/plugins\Mergin\mergin\deps\pygeodiff\geodifflib.py", line 328, in make_copy_sqlite
    _parse_return_code(res, "make_copy_sqlite")

  File "C:/Users/robwi/AppData/Roaming/QGIS/QGIS3\profiles\2019-10-27 New Profile for Q 3.1x/python/plugins\Mergin\mergin\deps\pygeodiff\geodifflib.py", line 41, in _parse_return_code
    raise GeoDiffLibError(msg)

Mergin.mergin.deps.pygeodiff.geodifflib.GeoDiffLibError: make_copy_sqlite
```

**Comment by rjwillson** [comment]
I have uploaded my client-log.txt for the failed sync yesterday and early today.

in the meantime I need to access the data I collected yesterday so I have used mergin to remove my local copy and then re-downloaded the project from the mergin server

I have also uploaded my successful sync that I just did with the re-downloaded project folder for comparison


[client-log - failed sync.txt](https://github.com/lutraconsulting/qgis-mergin-plugin/files/6856053/client-log.-.failed.sync.txt)
[client-log successful sync.txt](https://github.com/lutraconsulting/qgis-mergin-plugin/files/6856056/client-log.successful.sync.txt)

**Comment by wonder-sk** [comment]
This should be fixed now in the new release 2021.4.1 thanks to the upgrade of geodiff to 1.0.2 which fixes the underlying issue https://github.com/lutraconsulting/geodiff/issues/138
